samples/subsys/logging/logger/src/main.c

```diff
@@ -218,21 +218,22 @@ void main(void)
 		       log_source_id_get(INST2_NAME),
 		       CONFIG_LOG_DEFAULT_LEVEL);
 
-	while (true == LOG_PROCESS()) {
-	}
+	/* Ensuring that log thread will process. */
+	k_sleep(sleep_period);
 
 	severity_levels_showcase();
-
-	while (true == LOG_PROCESS()) {
-	}
+	/* Ensuring that log thread will process. */
+	k_sleep(sleep_period);
```

Comment:
Could we have a log_flush() or something else which will block current thread until log queue is empty?

Reply:
I've added log_buffered_cnt() to log_ctrl.h api which returns number of currently pending log messages. It is later on used in the example to pend on log flush